File: tests/Generator/resource/python/complex/map_property_type.py
from pydantic import BaseModel, Field, GetCoreSchemaHandler, Tag
from pydantic_core import CoreSchema, core_schema
from typing import Any, Dict, Generic, List, Optional, TypeVar, Annotated, Union, Literal
from .collection_property_type import CollectionPropertyType


# Represents a map which contains a dynamic set of key value entries of the same type
class MapPropertyType(CollectionPropertyType):
    type: Literal["map"] = Field(alias="type")
    pass

File: tests/Generator/resource/java/complex/MapPropertyType.java
package org.typeapi.model;

import com.fasterxml.jackson.annotation.*;

@JsonClassDescription("Represents a map which contains a dynamic set of key value entries of the same type")
public class MapPropertyType extends CollectionPropertyType {
    private String type = "map";

    @JsonSetter("type")
    public void setType(String type) {
        this.type = type;
    }

    @JsonGetter("type")
    public String getType() {
        return this.type;
    }
}

